R/forecastingdata_org.R

```diff
@@ -0,0 +1,380 @@
+#' @param key The name of the attribute that should be used as the key when creating the tsibble. The default value is "series_name" which is the key of the datasets in our repository. If doesn't provide, a data frame will be returned instead of a tsibble.
+#' @param value_column_name Any name that is preferred to have as the name of the column containing series values in the returning tsibble.
+#'
+#' @return This function returns a tsibble or a data.frame based on the validity of given key and index values.
+#' 
+#' @references 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Repository. \url{https://forecastingdata.org/}
+#' 
+#' Godahewa, R., Bergmeir, C., Webb, G. I., Hyndman, R. J. & Montero-Manso, P. (2021), Monash Time Series Forecasting Archive.
+#' 
+#' @export
+get_forecastingdata <- function(dataset=c("nn5", "nn5_without_missing", "nn5_weekly", "m1_yearly", 
+                                          "m1_quarterly", "m1_monthly", "m3_yearly", "m3_quarterly", 
+                                          "m3_monthly", "m3_other", "m4_yearly", "m4_quarterly", 
+                                          "m4_monthly", "m4_weekly", "m4_daily", "m4_hourly", 
+                                          "tourism_yearly", "tourism_quarterly", "tourism_monthly", 
+                                          "carparts", "carparts_without_missing", "hospital", "weather", 
+                                          "dominick", "fred_md", "solar_10_minutes", "solar_weekly", 
+                                          "solar_4_seconds", "wind_4_seconds", "sunspot", 
+                                          "sunspot_without_missing", "wind_farms", 
+                                          "wind_farms_without_missing", "elecdemand", "us_births", 
+                                          "saugeenday", "covid", "cif", "london_smart_meters", 
+                                          "london_smart_meters_without_missing", "web_traffic", 
+                                          "web_traffic_without_missing", "web_traffic_weekly", 
+                                          "traffic_hourly", "traffic_weekly", "electricity_hourly", 
+                                          "electricity_weekly", "pedestrians", "kdd", 
+                                          "kdd_without_missing", "aus_elecdemand", "covid_mobility_with_missing", 
+                                          "covid_mobility_without_missing"), 
+                                          destination_folder = '', index = NULL, key = "series_name", 
+                                          value_column_name = "series_value"){
+ 
+  dataset <- match.arg(dataset)
+ 
+  # Current datasets in archive
+  forecastingdata <- data.frame(nn5 = c("4656110", "nn5_daily_dataset_with_missing_values"),
```

Comment:
This can be stored as a separate non-exported object, so that it doesn't get created each time the function is used.

Reply:
Created forecastingdata data.frame at the beginning of the script.